### ISSUE TITLE
Invalid code generated when using both `deepImmutablesDetection` and `beanFriendlyModifiables`

### ISSUE THREAD

**Issue by lassemaatta** [immutables/immutables#619]
Example code:

```
@Value.Immutable
@Value.Modifiable
@Value.Style(
        deepImmutablesDetection = true,
        beanFriendlyModifiables = true
)
public abstract class Outer {
    public abstract List<Inner> getZeds();

    @Value.Immutable
    @Value.Modifiable
    public abstract static class Inner {
        public abstract String getBoo();
    }
}
```

generates invalid code for ModifiableOuter:

```
  public void addZeds(Outer.Inner element) {
    Outer.Inner nonNullElement = Preconditions.checkNotNull(element, "zeds element");
    zeds.add(element instanceof ModifiableInner ? element : ModifiableInner.create().from(element));
    return;
  }
```

because `.from()` has a return type of `void`.

Tested with 2.5.0

**Comment by augustotravillio** [comment]
Good catch, tnx! will be released in 2.5.1

**Comment by lassemaatta** [comment]
Thanks for your quick response! Looking forward to 2.5.1, keep up the good work :)
